omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/saga/omega/connector/grpc/LoadBalancedClusterMessageSender.java

```diff
@@ -53,20 +53,26 @@
   private final Collection<ManagedChannel> channels;
 
   private final BlockingQueue<Runnable> pendingTasks = new LinkedBlockingQueue<>();
-  private final BlockingQueue<MessageSender> availableMessageSenders = new LinkedBlockingQueue<>();
+  private final BlockingQueue<MessageSender> availableMessageSenders;
+  private final MessageSender retryableMessageSender;
   private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
 
   public LoadBalancedClusterMessageSender(String[] addresses,
       MessageSerializer serializer,
       MessageDeserializer deserializer,
       ServiceConfig serviceConfig,
       MessageHandler handler,
-      int reconnectDelay) {
+      int reconnectDelay,
+      BlockingQueue<MessageSender> availableMessageSenders,
+      MessageSender retryableMessageSender) {
 
     if (addresses.length == 0) {
       throw new IllegalArgumentException("No reachable cluster address provided");
     }
 
+    this.availableMessageSenders = availableMessageSenders;
+    this.retryableMessageSender = retryableMessageSender;
```

Comment:
why not construct retryable sender here to avoid passing both blocking queue and sender in?